packages/http/src/__tests__/http-prism-instance.spec.ts

```diff
@@ -1,114 +1,268 @@
-    });
+    test('when processed w/o baseUrl ignore server validation', async () => {
+      const result = await prism.process({
+        method: 'get',
+        url: {
+          path: '/pet',
+        },
+      });
 
-    expect(prism.resources).toStrictEqual(second_prism.resources);
-  });
+      expect(result.output!.statusCode).toBe(200);
+    });
 
-  test('given incorrect route should throw error', () => {
-    return expect(
-      prism.process({
+    test('when processed with valid baseUrl should return 200', async () => {
+      const result = await prism.process({
         method: 'get',
         url: {
-          path: '/invalid-route',
+          path: '/pet',
+          baseUrl: 'http://example.com/api',
         },
-      }),
-    ).rejects.toThrowError(ProblemJsonError.fromTemplate(NO_PATH_MATCHED_ERROR));
-  });
+      });
 
-  test('given correct route should return correct response', async () => {
-    const response = await prism.process({
-      method: 'get',
-      url: {
-        path: '/pet/findByStatus',
-        query: {
-          status: ['available', 'pending'],
-        },
-      },
+      expect(result.output!.statusCode).toBe(200);
     });
 
-    const parsedBody = response!.output!.body;
-
-    expect(parsedBody.length).toBeGreaterThan(0);
+    test('when processed with invalid host should throw error', () => {
+      return expect(
+        prism.process({
+          method: 'get',
+          url: {
+            path: '/pet',
+            baseUrl: 'http://acme.com/api',
+          },
+        }),
+      ).rejects.toThrowError(ProblemJsonError.fromTemplate(NO_SERVER_MATCHED_ERROR));
+    });
 
-    parsedBody.forEach((element: any) => {
-      expect(typeof element.name).toBe('string');
-      expect(element.photoUrls).toBeInstanceOf(Array);
-      expect(element.photoUrls.length).toBeGreaterThan(0);
+    test('when processed with invalid baseUrl should throw error', () => {
+      return expect(
+        prism.process({
+          method: 'get',
+          url: {
+            path: '/pet',
+            baseUrl: 'http://example.com/v1',
+          },
+        }),
+      ).rejects.toThrowError(ProblemJsonError.fromTemplate(NO_SERVER_MATCHED_ERROR));
     });
 
-    expect(response).toMatchSnapshot({
-      output: {
-        body: expect.anything(),
-      },
+    describe('given mocking is turned off', () => {
+      const config: IHttpConfig = { mock: false };
+      const baseUrl = 'http://stoplight.io';
+      const serverReply = 'hello world';
+
+      beforeEach(() => {
+        nock(baseUrl)
+          .get('/x-bet')
+          .reply(200, serverReply);
+      });
+
+      afterEach(() => {
+        nock.cleanAll();
+      });
+
+      describe('given invalid path', () => {
+        const request: IHttpRequest = {
+          method: 'get',
+          url: {
+            path: '/x-bet',
+            baseUrl,
+          },
+        };
+
+        test('returns input warning', async () => {
```

Comment:
so this `test` doesn't have `when` or `given`

Reply:
I considered 'when request processed then returns input warning' to be too verbose.
Your thoughts?

Reply:
I was only about inconsistency, the other `tests` have `when`.

Reply:
Yup, spot on @lag-of-death! Commit is on its way.